src/plugins/console/public/lib/osd/api.ts

```diff
@@ -44,41 +46,50 @@ import {
  * @constructor
  * @param bodyParametrizedComponentFactories same as urlParametrizedComponentFactories but used for body compilation
  */
-function Api(urlParametrizedComponentFactories, bodyParametrizedComponentFactories) {
-  this.globalRules = Object.create(null);
-  this.endpoints = Object.create(null);
-  this.urlPatternMatcher = new UrlPatternMatcher(urlParametrizedComponentFactories);
-  this.globalBodyComponentFactories = bodyParametrizedComponentFactories;
-  this.name = '';
-}
+export class Api {
+  globalRules: Record<string, any>;
+  endpoints: Record<string, Endpoint>;
+  urlPatternMatcher: UrlPatternMatcher;
+  globalBodyComponentFactories: ParametrizedComponentFactories | undefined;
+  name: string;
+
+  constructor(
+    urlParametrizedComponentFactories?: ParametrizedComponentFactories,
+    bodyParametrizedComponentFactories?: ParametrizedComponentFactories
+  ) {
+    this.globalRules = Object.create(null);
+    this.endpoints = Object.create(null);
+    this.urlPatternMatcher = new UrlPatternMatcher(urlParametrizedComponentFactories);
+    this.globalBodyComponentFactories = bodyParametrizedComponentFactories;
+    this.name = '';
+  }
 
-(function (cls) {
-  cls.addGlobalAutocompleteRules = function (parentNode, rules) {
+  addGlobalAutocompleteRules(parentNode: string, rules: Record<string, any>): void {
     this.globalRules[parentNode] = compileBodyDescription(
       'GLOBAL.' + parentNode,
       rules,
       this.globalBodyComponentFactories
     );
-  };
+  }
 
-  cls.getGlobalAutocompleteComponents = function (term, throwOnMissing) {
-    const result = this.globalRules[term];
+  getGlobalAutocompleteComponents(term: string, throwOnMissing?: boolean) {
+    const result: SharedComponent[] = this.globalRules[term];
```

Comment:
`globalRules` is defined as `Record<string, any>`. `Record<string, any>` implies that the values in `globalRules` could be of any type, so we can't directly assert that this.globalRules[term] is of type SharedComponent[], right?

Reply:
Yes, that's right. I referred to the type defined in file that already was in TS.
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/487cd363abd91559369bb9c8fc7771175265bd0d/src/plugins/console/server/services/spec_definitions_service.ts#L43-L45